File: solution/415. Add Strings/415.java
class Solution {
    public String addStrings(String num1, String num2) {
        int s1 = num1.length() - 1, s2 = num2.length() - 1;
        int carry = 0;
        StringBuilder sb = new StringBuilder();
        
        while (s1 >= 0 || s2 >= 0) {
            int n1 = s1 >= 0 ? num1.charAt(s1--) - '0' : 0;
            int n2 = s2 >= 0 ? num2.charAt(s2--) - '0' : 0;
            int sum = n1 + n2 + carry;
            sb.append(sum % 10);
            carry = sum / 10;
        }
        String res = sb.reverse().toString();
        return carry != 0 ? "1" + res : res;
    }
}

File: solution/415. Add Strings/415.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int s1 = num1.size() - 1, s2 = num2.size() - 1;
        string res = "";
        int carry = 0;
        
        while (s1 >= 0 || s2 >= 0) {
            int n1 = s1 >= 0 ? num1[s1--] - '0' : 0;
            int n2 = s2 >= 0 ? num2[s2--] - '0' : 0;
            int sum = n1 + n2 + carry;
            res.insert(res.begin(), sum % 10 + '0');
            carry = sum / 10;
        }
        return carry ? "1" + res : res;
    }
};
